src/Testing/src/xunit/QuarantinedTestTraitDiscoverer.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+// Do not change this namespace without changing the usage in QuarantinedTestAttribute
+namespace Microsoft.AspNetCore.Testing
+{
+    public class QuarantinedTestTraitDiscoverer : ITraitDiscoverer
+    {
+        public IEnumerable<KeyValuePair<string, string>> GetTraits(IAttributeInfo traitAttribute)
+        {
+            if (traitAttribute is ReflectionAttributeInfo attribute && attribute.Attribute is QuarantinedTestAttribute quarantinedTestAttribute)
+            {
+                yield return new KeyValuePair<string, string>("Quarantined", "true");
+            }
+            else
+            {
+                throw new InvalidOperationException("The 'QuarantinedTest' attribute is only supported via reflection.");
```

Comment:
See this was here before. Does it mean we can't use other traits in the future w/o changing this `ITraitDiscoverer`?

Reply:
🤷‍♂ Not familiar with this, and didn't change the code.

Reply:
The code will `throw` if another `traitAttribute` is passed in. But, the more I think about it, the less this matters. Just something to keep in mind if we decide to use additional traits.

Reply:
> See this was here before. Does it mean we can't use other traits in the future w/o changing this `ITraitDiscoverer`?

No, it's because xUnit technically abstracts over the use of attributes entirely. The idea that we'd ever have an attribute that isn't really a .NET attribute (readable via reflection) is extremely unlikey but one that xUnit requires you to consider (via downcasting).

Technically it only really mattered when we were reading properties on the attribute (to get the "FlakyOn" filters) but I can't even fathom a case where we'd want this code path to run (we'd be doing some very weird xUnit stuff if it did) so throwing an exception here is reasonable.

Reply:
My point was about the `QuarantinedTestAttribute` cast, not the `ReflectionAttributeInfo` one. If we add another `ITraitAttribute` implementation, will reach here and `throw`. But, again, this is not a big deal right now.

Reply:
That would require that the other attribute has designated this Discoverer. Each attribute explicitly specifies which Discoverer to use.